syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.snippet_sample";
option java_outer_classname = "snippet_sample_Proto";
option objc_class_prefix = "SSP";

package StorageEngineInstance;

service MonitoringContainer {
  rpc SetCSDMetricsInfo (CSDMetricList) returns (Result) {}
}

message CSDMetricList{
  message CSDMetric{
    string id = 1;
    string ip = 2;
    float cpu_usage = 3;
    float memory_usage = 4;
    float disk_usage = 5;
    float network = 6;
    int32 working_block_count = 7;
  }
  repeated CSDMetric csd_metric_list = 1;
}

message Result{
  string message = 1;
}